src/QMCWaveFunctions/LCAO/LCAOrbitalSetT.h

```diff
@@ -0,0 +1,336 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by:
+//
+// File created by: Jeongnim Kim, jeongnim.kim@intel.com, Intel Corp.
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_SOA_LINEARCOMIBINATIONORBITALSET_TEMP_H
+#define QMCPLUSPLUS_SOA_LINEARCOMIBINATIONORBITALSET_TEMP_H
+
+#include <memory>
+#include "QMCWaveFunctions/SPOSetT.h"
+#include "QMCWaveFunctions/BasisSetBase.h"
+
+#include "Numerics/MatrixOperators.h"
+#include "Numerics/DeterminantOperators.h"
+
+namespace qmcplusplus
+{
+/** class to handle linear combinations of basis orbitals used to evaluate the Dirac determinants.
+   *
+   * SoA verson of LCOrtbitalSet
+   * Localized basis set is always real 
+   */
+template<class T>  
```

Comment:
There is a possibility that some of the arguments to SPOSet functions should always be `FullPrec` So do we want to revisit that here? There should be a VALUE and VALUE_FP argument then. 

I think here we should maintain the menomic trace that this is a Value type that is being templated on here.  So use VALUE or VALUE_FP for the template parameter.  Since there are rougly speaking two categories of templating on scalars in QMCPACK.  There are scalars that historically are called values can be real/complex reduced/full and reals those that are real reduced/full.

Reply:
@PDoakORNL Would you repeat this question about `SPOSetT<T>` on the main [QMCPACK PR](https://github.com/QMCPACK/qmcpack/pull/4684)?